docs/en/sql-reference/table-functions/s3.md

```diff
@@ -43,35 +43,42 @@ A table with the specified structure for reading or writing data in the specifie
 
 **Examples**
 
-Selecting the first two rows from the table from S3 file `https://clickhouse-public-datasets.s3.amazonaws.com/my-test-bucket-768/data.csv`:
+Selecting the first 5 rows from the table from S3 file `https://datasets-documentation.s3.eu-west-3.amazonaws.com/aapl_stock.csv`:
 
 ``` sql
 SELECT *
-FROM s3('https://clickhouse-public-datasets.s3.amazonaws.com/my-test-bucket-768/data.csv', 'CSV', 'column1 UInt32, column2 UInt32, column3 UInt32')
-LIMIT 2;
+FROM s3(
+   'https://datasets-documentation.s3.eu-west-3.amazonaws.com/aapl_stock.csv',
+   'CSVWithNames'
+)
+LIMIT 5;
 ```
 
-``` text
-┌─column1─┬─column2─┬─column3─┐
-│       1 │       2 │       3 │
-│       3 │       2 │       1 │
-└─────────┴─────────┴─────────┘
+```response
+┌───────Date─┬────Open─┬────High─┬─────Low─┬───Close─┬───Volume─┬─OpenInt─┐
+│ 1984-09-07 │ 0.42388 │ 0.42902 │ 0.41874 │ 0.42388 │ 23220030 │       0 │
+│ 1984-09-10 │ 0.42388 │ 0.42516 │ 0.41366 │ 0.42134 │ 18022532 │       0 │
+│ 1984-09-11 │ 0.42516 │ 0.43668 │ 0.42516 │ 0.42902 │ 42498199 │       0 │
+│ 1984-09-12 │ 0.42902 │ 0.43157 │ 0.41618 │ 0.41618 │ 37125801 │       0 │
+│ 1984-09-13 │ 0.43927 │ 0.44052 │ 0.43927 │ 0.43927 │ 57822062 │       0 │
+└────────────┴─────────┴─────────┴─────────┴─────────┴──────────┴─────────┘
 ```
 
-The similar but from file with `gzip` compression:
+:::note
+ClickHouse uses filename extensions to determine the format of the data and also the compression of the file. For example, if the file was zipped up with a `.csv.gz` extension, ClickHouse would decompress the file.
+
+To demonstrate, try the following command - which works the same as the example above:
 
 ``` sql
 SELECT *
-FROM s3('https://clickhouse-public-datasets.s3.amazonaws.com/my-test-bucket-768/data.csv.gz', 'CSV', 'column1 UInt32, column2 UInt32, column3 UInt32', 'gzip')
-LIMIT 2;
+FROM s3(
+   'https://datasets-documentation.s3.eu-west-3.amazonaws.com/aapl_stock.csv'
```

Comment:
seems like it should be like
```
FROM s3(
    'https://datasets-documentation.s3.eu-west-3.amazonaws.com/aapl_stock.csv.gz'
```
instead

Reply:
Oh...but then the command won't work because there is no file in S3 with the .gz extension. I'm just showing here that you can leave off the format. I will reword it...